### PR TITLE
Add some additional tests for storage/types.go

### PR DESCRIPTION
**Pull request by Martin2112**
Strengthen the tests for Equivalent NodeIDs. Add new tests and also check that n1.Equivalent(n2) matches n2.Equivalent(n1). Boost code coverage for storage/types.go.

New tests: 

* Check that 'nearby' in log coordinate tree node IDs don't compare equal but same ones do at 8 levels, exercising all 'leftover' number of bits.
* Construct some ids with 56 significant bits out of 64, mess with the last 8 bits.
* Test NewNodeIDFromHash() and Equivalent() on different length input.
* Add tests for all the cases where the NodeID code should panic on bad input.
* Add tests for a couple of methods that are used but weren't tested.
* Fix a couple of typos in the doc strings for types.go.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by Martin2112** [comment]
Normally I don't like writing tests to boost code coverage but this was leaving a lot of edge cases untested.

File: storage/types.go [hunk 1]
```diff
@@ -318,10 +318,10 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, Suffix) {
 
 	bits := n.PrefixLenBits - prefixBytes*8
 	if bits > suffixBits {
-		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v *8) > %v", n.Path, n.PrefixLenBits, prefixBytes, suffixBits))
+		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v*8) > %v", n.Path, n.PrefixLenBits, prefixBytes, suffixBits))
 	}
 	if bits == 0 {
```
Comment (daviddrysdale) [inline storage/types.go]: Might this be better as `bits <= 0`?  Currently your added test for this case (`TestSplitPanics/splitoutofbounds`) panics with `slice bounds out of range` from line 329 below.

File: storage/types_test.go [hunk 2]
```diff
@@ -147,6 +274,70 @@ func TestSplit(t *testing.T) {
...
+					t.Errorf("Incorrect panic behaviour got: %v, want: %v", got, tc.wantPanic)
+				}
+			}()
+			_, _ = n.Split(tc.pBytes, tc.sBits)
+		})
+	}
+	// The 2 cases where split should panic are 1.) There are more bits left
+	// after the prefix point than have been requested. 2.) Trying to split zero
+	// bits off the end of the NodeID.
+	for b := 0; b < 64; b++ {
```
Comment (daviddrysdale) [inline storage/types_test.go]: This comment is helpful but the loop isn't....

File: storage/types.go (hunk 1 above)
Comment (Martin2112) [reply storage/types.go]: Done.

File: storage/types_test.go (hunk 2 above)
Comment (Martin2112) [reply storage/types_test.go]: Oops. Didn't clean up properly after restructure. Fixed.

File: storage/types.go [hunk 3]
```diff
@@ -320,7 +320,7 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, Suffix) {
 	if b > suffixBits {
 		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v *8) > %v", n.Path, n.PrefixLenBits, prefixBytes, suffixBits))
 	}
-	if b == 0 {
+	if b <= 0 {
 		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v *8) == 0", n.Path, n.PrefixLenBits, prefixBytes))
```
Comment (daviddrysdale) [inline storage/types.go]: Panic message needs an update too.

Comment (Martin2112) [reply storage/types.go]: Done.
